src/status_im/contexts/wallet/home/view.cljs

```diff
@@ -16,19 +17,22 @@
 
 (defn new-account
   []
-  [quo/action-drawer
-   [[{:icon                :i/add
-      :accessibility-label :start-a-new-chat
-      :label               (i18n/label :t/add-account)
-      :sub-label           (i18n/label :t/add-account-description)
-      :on-press            #(rf/dispatch [:navigate-to :screen/wallet.create-account])}
-     (when (ff/enabled? ::ff/wallet.add-watched-address)
-       {:icon                :i/reveal
-        :accessibility-label :add-a-contact
-        :label               (i18n/label :t/add-address-to-watch)
-        :sub-label           (i18n/label :t/add-address-to-watch-description)
-        :on-press            #(rf/dispatch [:navigate-to :screen/wallet.add-address-to-watch])
-        :add-divider?        true})]]])
+  (let [watched-accounts             (rf/sub [:wallet/watch-only-accounts])
+        reached-max-watched-account? (>= (count watched-accounts)
+                                         constants/max-allowed-watched-accounts)]
+    [quo/action-drawer
+     [[{:icon                :i/add
+        :accessibility-label :start-a-new-chat
+        :label               (i18n/label :t/add-account)
+        :sub-label           (i18n/label :t/add-account-description)
+        :on-press            #(rf/dispatch [:navigate-to :screen/wallet.create-account])}
+       (when (not reached-max-watched-account?)
```

Comment:
Just a UI/UX thing: instead of removing the option to add watch address if the limit is exhausted. We should make it in disabled state or something. When the user taps on it, we need to inform them (a toast works too) that they can add only 1 account and if they need to add, they need to delete any existing one.

Reply:
We can have the check in this PR, seems simple enough !

Reply:
@shivekkhurana, check is fine, but about message if we are going to display i need a clear toast message, or i can just use some ai text.